File: src/Web/Models/Map/Map.cs
using System.Collections.Generic;

namespace WebSPA.Models.Map
{
    public class Map
    {
        public string Id { get; set; }
        public string Caption { get; set; }
        public string? Description { get; set; }

        public virtual IList<MapPath> Paths { get; set; }
        public virtual IList<City> Cities { get; set; }
    }
}

File: src/Web/src/Web.SPA/Client/src/shared/infastructure/models/backend/Map.ts
export type Position = {
    x: number;
    y: number;
}

export type Indicator = {
    position: Position;
    style?: string;
    size?: number;
}
export type Text = {
    caption: string;
    style?: string;
    position?: Position;
    align?: 'top' | 'bottom' | 'right'; // Cannot use 'left' because we don't know the actual length of caption
    size?: number;
}

export type City = {
    text: Text;
    indicator: Indicator;
}

export type Path = {
    value: string;
    style?: string;
}

export interface MapInfo {
    id: number;
    caption: string;
    description?: string;

    paths: Path[];
    cities: City[];
}

export const DEFAULT_INDICATOR_SIZE = 24;
export const DEFAULT_TEXT_SIZE = 24;